ui/kotlinx-coroutines-android/README.md

```diff
@@ -5,6 +5,63 @@ Provides `Dispatchers.Main` context for Android applications.
 Read [Guide to UI programming with coroutines](https://github.com/Kotlin/kotlinx.coroutines/blob/master/ui/coroutines-guide-ui.md)
 for tutorial on this module.
 
+# Optimization
+
+You can optimize the size of the coroutines library in Android projects by statically removing 
+debugging and other advanced features using R8 by adding the following rule to your 
+`proguard-rules.pro` file:
```

Comment:
Why not bundle these rules directly in `META_INF` so these optimizations are turned on by default on all projects, without manual configuration?

Reply:
Some people want to turn on debugging facilities. I'm not sure what would be the right default behavior and how configuration can be undone if we bundle it.

Reply:
It's very uncommon to enable R8 (using minifyEnabled) on a debug build.
I can think of someone wanting to debug coroutines post R8 processing though.

@wojtek-kalicinski are you aware of way to disable third-party rules for such a case?

Reply:
we're working on a method to do it, but it won't be ready soon.

I'm wondering if just putting an opposite `-assumevalues ... return true;` in your project's proguard config would work, i.e. if it would be read after the library one and override it. I'd need to test that to be sure.

In any case, I also prefer rules to be applied automatically, otherwise 90% developers won't do it. It would be ideal if disabling the debug stripping would be possible though.

Reply:
I'm bundling _some_ of the rules (like turning off debug support). It looks like it would be Ok to say that you need to turn off optimization to support debugging.

Reply:
I'm bundling all the rules in this PR.